### PR TITLE
Send auto_update config for update and action report

### PR DESCRIPTION
**Pull request by gabriel**
As discussed in this PR here: https://github.com/keybase/go-updater/pull/40#discussion_r61696909
Is also in design doc.

### CODE REVIEW

**Comment by gabriel** [comment]
@keybase/updater-hackers

File: keybase/report.go [hunk 1]
```diff
@@ -52,6 +52,8 @@ func (c context) reportAction(action updater.UpdateAction, options updater.Updat
 	data.Add("version", options.Version)
 	data.Add("upd_version", options.UpdaterVersion)
 	data.Add("action", action.String())
+	autoUpdate, _ := c.config.GetUpdateAuto()
```
Comment (maxtaco) [inline keybase/report.go]: If we've made the change on this iteration through the loop, is it reflected in this getter?

Comment (gabriel) [inline keybase/report.go]: yeah, the autoUpdate config is set/saved immediately after the prompt returns... and the action is reported after this... so we should be all good... I need to add more tests for this though

Comment (gabriel) [inline keybase/report.go]: Added test

**Comment by jzila** [comment]
lgtm

**Pull request merged**